src/modules/Dropdown/Dropdown.js

```diff
@@ -46,20 +46,11 @@ export default class Dropdown extends Component {
       PropTypes.string,
     ]),
 
-    /** Array of `{ text: '', value: '' }` options */
+    /** Array of Dropdown.Item props e.g. `{ text: '', value: '' }` */
     options: customPropTypes.every([
       customPropTypes.disallow(['children']),
       customPropTypes.demand(['selection']),
-      PropTypes.arrayOf(PropTypes.shape({
-        value: PropTypes.oneOfType([
-          PropTypes.string,
-          PropTypes.number,
-        ]).isRequired,
```

Comment:
Seems reasonable for now.  The text/value props can't reliably be required at the DropdownItem level as there is no guarantee they are being used in a selection Dropdown.